pkg/storage/tsdb/config.go

```diff
@@ -183,6 +184,7 @@ type BucketStoreConfig struct {
 // RegisterFlags registers the BucketStore flags
 func (cfg *BucketStoreConfig) RegisterFlags(f *flag.FlagSet) {
 	cfg.IndexCache.RegisterFlagsWithPrefix(f, "experimental.tsdb.bucket-store.index-cache.")
+	cfg.CachingBucket.RegisterFlagsWithPrefix(f, "experimental.tsdb.bucket-store.caching-bucket.")
```

Comment:
[not a blocker] This is a discussion we already had. I think that for the use it's a bit inconsistent seeing `index-cache` and `caching-bucket`. Technically we're currently having only chunks caching and even when we'll cache metadata we already agreed that we want to store metadata on different memcached server (at least giving the possibility to do it) so I'm wondering if in the Cortex implementation we could go more specific with `chunks-cache`.

_Let's talk about this offline_

Reply:
I agree with your naming. I think we also need to rethink usage of Thanos’ config object, and introduce our own config object here –– one for chunks, one for metadata. (As a consequence of https://github.com/thanos-io/thanos/pull/2579, even if not merged yet, I already think it will break stuff).

Reply:
Agree on using our own structs. We shouldn't directly use Thanos config structs, while we should use our own which wraps Thanos ones, in order to keep control of our config (ie. avoid config to change whenever Thanos renames stuff, or keep naming more consistent within Cortex).